oauth2_http/java/com/google/auth/oauth2/IdentityPoolCredentialSource.java

```diff
@@ -48,6 +50,157 @@ public class IdentityPoolCredentialSource extends ExternalAccountCredentials.Cre
+          String.format(
+              "Invalid type for '%s' in certificate configuration: expected String, got %s.",
+              key, value.getClass().getSimpleName()));
+    }
+    return (String) value;
+  }
+  /**
+   * Represents the configuration options for X.509-based workload credentials (mTLS). It specifies
+   * how to locate and use the client certificate, private key, and optional trust chain for mutual
+   * TLS authentication.
+   */
+  public static class CertificateConfig implements java.io.Serializable {
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * If true, attempts to load the default certificate configuration. It checks the
+     * GOOGLE_API_CERTIFICATE_CONFIG environment variable first, then a conventional default file
+     * location. Cannot be true if {@code certificateConfigLocation} is set.
+     */
+    private final boolean useDefaultCertificateConfig;
+
+    /**
+     * Specifies the path to the client certificate and private key file. This is used when {@code
+     * useDefaultCertificateConfig} is false or unset. Must be set if {@code
+     * useDefaultCertificateConfig} is false.
+     */
+    @Nullable private final String certificateConfigLocation;
+
+    /**
+     * Specifies the path to a PEM-formatted file containing the X.509 certificate trust chain. This
+     * file should contain any intermediate certificates required to complete the trust chain
+     * between the leaf certificate (used for mTLS) and the root certificate(s) in your workload
+     * identity pool's trust store. The leaf certificate and any certificates already present in the
+     * workload identity pool's trust store are optional in this file. Certificates should be
+     * ordered with the leaf certificate (or the certificate which signed the leaf) first.
+     */
+    @Nullable private final String trustChainPath;
+
+    /**
+     * Constructor for {@code CertificateConfig}.
+     *
+     * @param useDefaultCertificateConfig Whether to use the default certificate configuration.
+     * @param certificateConfigLocation Path to the client certificate and private key file.
+     * @param trustChainPath Path to the trust chain file.
+     * @throws IllegalArgumentException if the configuration is invalid (e.g., neither default nor
+     *     location is specified, or both are specified).
+     */
+    CertificateConfig(
+        @Nullable Boolean useDefaultCertificateConfig,
+        @Nullable String certificateConfigLocation,
+        @Nullable String trustChainPath) {
+
+      boolean useDefault = useDefaultCertificateConfig != null && useDefaultCertificateConfig;
+      boolean locationIsPresent =
+          certificateConfigLocation != null && !certificateConfigLocation.isEmpty();
+
+      checkArgument(
+          !(!useDefault && !locationIsPresent),
+          "credentials: \"certificate\" object must either specify a certificate_config_location or use_default_certificate_config should be true");
```

Comment:
nit: I think this error message may be a bit confusing since users may not know where the `certificate object` comes from.

Would it be accurate to say something like `The certificate source must specify a certificate_config_location value or set the use_default_certificate_config value to be true`?

Reply:
The error uses the exact wording as our Go implementation, but I can change it to certificate "source" if you think it'd be more clear for the users.

Reply:
I see, I will leave it up to you if consistency among languages is a concern.

My preference would be for the error message be more actionable. I think the wording above may be a bit unclear for a first time user as they may not know where the `certificate object` is coming from. If possible, could we specify that the certificate object is from the `certificate source` (or even something more specific)?

Reply:
Updated the error to: "Invalid 'certificate' configuration in credential source: Must specify either 'certificate_config_location' or set 'use_default_certificate_config' to true."
Let me know what you think!